### PR TITLE
Bake HuggingFace models into DVC + Docker

### PR DESCRIPTION
**Pull request by mdekstrand**
This adds the HuggingFace models directly to our DVC repo, bakes them into the Docker image, and loads them from there instead of directly from HF behind the scenes.

This improves the situation in a few ways:

* Our HuggingFace models are version-locked so they can't change underneath us.
* All resources are tracked directly instead of fetched through behind-the-scenes HTTP downloads.
* Startup is probably faster because it doesn't need to re-fetch a half-gig of model data from HF.

### CODE REVIEW

File: src/poprox_recommender/model/__init__.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from poprox_recommender.model.nrms import NRMS
 from poprox_recommender.paths import model_file_path
 
+TOKENIZER_NAME = "distilbert-base-uncased"
```
Comment (karlhigley) [inline src/poprox_recommender/model/__init__.py:12]:
```suggestion
MODEL_NAME = "distilbert-base-uncased"
```

File: src/poprox_recommender/model/__init__.py [hunk 2]
```diff
@@ -26,7 +27,7 @@ class ModelConfig:
     num_attention_heads: float = 16
     hidden_size: int = 768
 
-    pretrained_model = "distilbert-base-uncased"
+    pretrained_model = TOKENIZER_NAME
```
Comment (karlhigley) [inline src/poprox_recommender/model/__init__.py:30]:
```suggestion
    pretrained_model = MODEL_NAME
```

File: src/poprox_recommender/model/__init__.py [hunk 3]
```diff
@@ -60,7 +61,7 @@ def load_model(checkpoint, device):
 def get_model(device=None) -> RecommenderComponents:
     global _cached_model
     if _cached_model is None:
-        tokenizer = AutoTokenizer.from_pretrained("distilbert-base-uncased", cache_dir="/tmp/")
+        tokenizer = AutoTokenizer.from_pretrained(model_file_path(TOKENIZER_NAME), cache_dir="/tmp/")
```
Comment (karlhigley) [inline src/poprox_recommender/model/__init__.py:64]:
```suggestion
        tokenizer = AutoTokenizer.from_pretrained(model_file_path(MODEL_NAME), cache_dir="/tmp/")
```

File: src/poprox_recommender/topics.py [hunk 4]
```diff
@@ -6,6 +6,9 @@
 
 from poprox_concepts import Article, ClickHistory
 from poprox_concepts.domain.topics import GENERAL_TOPICS
+from poprox_recommender.paths import model_file_path
+
+NEWS_MODEL = "news-category-classifier-distilbert"
```
Comment (karlhigley) [inline src/poprox_recommender/topics.py:11]: Though the code for topic matching exists in the repo, I don't think we're using it at all right now. Baking this model into the container image is an extra 400MB we don't currently need AFAIK

**Review by karlhigley — APPROVED** [review]

File: src/poprox_recommender/topics.py (hunk 4 above)
Comment (mdekstrand) [reply src/poprox_recommender/topics.py]: This worked.

**Comment by mdekstrand** [comment]
@karlhigley Just merged this, see Slack for one outstanding issue the final testing run-up surfaced.

**Pull request merged**